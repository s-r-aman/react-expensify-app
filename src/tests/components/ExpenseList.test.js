import React from 'react';
import { shallow } from "enzyme";

import expenses from "../fixtures/expenses";
import { ExpenseList } from "../../components/ExpenseList";

describe('<ExpenseList/>', () => {
  test('should render the expenselist correctly with some expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
  });
  test("should render the expenselist correctly without any expenses", () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});