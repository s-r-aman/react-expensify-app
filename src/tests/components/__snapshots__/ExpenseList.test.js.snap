// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseList/> should render the expenselist correctly with some expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="Rent"
    id="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    note=""
  />
</div>
`;

exports[`<ExpenseList/> should render the expenselist correctly without any expenses 1`] = `
<div>
  <p>
    No expenses.
  </p>
</div>
`;
